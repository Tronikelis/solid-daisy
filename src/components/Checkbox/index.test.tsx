/* eslint-disable solid/reactivity */
import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it } from "vitest";

import { Checkbox } from ".";

describe("<Checkbox />", () => {
    it("renders correctly", () => {
        const { container } = render(() => <Checkbox />);
        expect(container).toMatchSnapshot();
    });

    it("controlled input works", async () => {
        const [checked, setChecked] = createSignal(false);

        const screen = render(() => (
            <Checkbox
                data-testid="checkbox"
                checked={checked()}
                onInput={e => setChecked(e.target.checked)}
            />
        ));

        expect(screen.getByTestId("checkbox")).not.toBeChecked();

        await userEvent.click(screen.getByTestId("checkbox"));

        expect(checked()).toBe(true);
        expect(screen.getByTestId("checkbox")).toBeChecked();

        setChecked(false);

        expect(screen.getByTestId("checkbox")).not.toBeChecked();
    });
});
