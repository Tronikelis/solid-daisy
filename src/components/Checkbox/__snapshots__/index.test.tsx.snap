// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Checkbox /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-4"
  >
    <label
      class="label cursor-pointer gap-3"
    >
      <input
        class="checkbox"
        type="checkbox"
      />
      
    </label>
  </div>
</div>
`;
